From 7183b80926cd25cea35b08b43cde49dede3eb467 Mon Sep 17 00:00:00 2001
From: Kevin Hao <kexin.hao@windriver.com>
Date: Thu, 1 Sep 2011 16:43:59 +0800
Subject: [PATCH] powerpc/p2020: enable two workarounds for eSDHC

Enable two workarounds for eSDHC on p2020 board.
- DMA broken.
- bit DCR[DMA__AHB2MAG_IRQ_BYPASS] can't be set automatically when
  SoC reset.

This is based on the patch "Enable eSDHC feature on fsl_p2020ds
board" in FSL P2020DS_20091110_ltib.iso vendor drop.

Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/p2020ds.dts                 |    2 ++
 arch/powerpc/boot/dts/p2020ds_36b.dts             |    2 ++
 arch/powerpc/boot/dts/p2020ds_camp_core0_36b.dts  |    2 ++
 arch/powerpc/boot/dts/p2020rdb.dts                |    1 +
 arch/powerpc/boot/dts/p2020rdb_36b.dts            |    2 ++
 arch/powerpc/boot/dts/p2020rdb_camp_core0.dts     |    1 +
 arch/powerpc/boot/dts/p2020rdb_camp_core0_36b.dts |    2 ++
 7 files changed, 12 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/boot/dts/p2020ds.dts b/arch/powerpc/boot/dts/p2020ds.dts
index 491159b..96917f0 100644
--- a/arch/powerpc/boot/dts/p2020ds.dts
+++ b/arch/powerpc/boot/dts/p2020ds.dts
@@ -549,6 +549,8 @@
 			reg = <0x2e000 0x1000>;
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
+			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			/* Filled in by U-Boot */
 			clock-frequency = <0>;
diff --git a/arch/powerpc/boot/dts/p2020ds_36b.dts b/arch/powerpc/boot/dts/p2020ds_36b.dts
index 964dd74..49001ba 100644
--- a/arch/powerpc/boot/dts/p2020ds_36b.dts
+++ b/arch/powerpc/boot/dts/p2020ds_36b.dts
@@ -508,6 +508,8 @@
 			reg = <0x2e000 0x1000>;
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
+			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			/* Filled in by U-Boot */
 			clock-frequency = <0>;
diff --git a/arch/powerpc/boot/dts/p2020ds_camp_core0_36b.dts b/arch/powerpc/boot/dts/p2020ds_camp_core0_36b.dts
index f9c7732..cbe8747 100644
--- a/arch/powerpc/boot/dts/p2020ds_camp_core0_36b.dts
+++ b/arch/powerpc/boot/dts/p2020ds_camp_core0_36b.dts
@@ -217,6 +217,8 @@
 			reg = <0x2e000 0x1000>;
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
+			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			clock-frequency = <0>;
 		};
diff --git a/arch/powerpc/boot/dts/p2020rdb.dts b/arch/powerpc/boot/dts/p2020rdb.dts
index 93dc6d9..9d9e1cb 100644
--- a/arch/powerpc/boot/dts/p2020rdb.dts
+++ b/arch/powerpc/boot/dts/p2020rdb.dts
@@ -504,6 +504,7 @@
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
 			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			/* Filled in by U-Boot */
 			clock-frequency = <0>;
diff --git a/arch/powerpc/boot/dts/p2020rdb_36b.dts b/arch/powerpc/boot/dts/p2020rdb_36b.dts
index 974d5c7..56eb15c 100644
--- a/arch/powerpc/boot/dts/p2020rdb_36b.dts
+++ b/arch/powerpc/boot/dts/p2020rdb_36b.dts
@@ -491,6 +491,8 @@
 			reg = <0x2e000 0x1000>;
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
+			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			/* Filled in by U-Boot */
 			clock-frequency = <0>;
diff --git a/arch/powerpc/boot/dts/p2020rdb_camp_core0.dts b/arch/powerpc/boot/dts/p2020rdb_camp_core0.dts
index 12f31db..57d9898 100644
--- a/arch/powerpc/boot/dts/p2020rdb_camp_core0.dts
+++ b/arch/powerpc/boot/dts/p2020rdb_camp_core0.dts
@@ -295,6 +295,7 @@
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
 			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			/* Filled in by U-Boot */
 			clock-frequency = <0>;
diff --git a/arch/powerpc/boot/dts/p2020rdb_camp_core0_36b.dts b/arch/powerpc/boot/dts/p2020rdb_camp_core0_36b.dts
index 9be3021..e643cf8 100644
--- a/arch/powerpc/boot/dts/p2020rdb_camp_core0_36b.dts
+++ b/arch/powerpc/boot/dts/p2020rdb_camp_core0_36b.dts
@@ -294,6 +294,8 @@
 			reg = <0x2e000 0x1000>;
 			interrupts = <72 0x2>;
 			interrupt-parent = <&mpic>;
+			fsl,sdhci-dma-broken;
+			fsl,sdhci-ahb2mag-irq-bypass;
 			fsl,sdhci-adjust-timeout;
 			/* Filled in by U-Boot */
 			clock-frequency = <0>;
-- 
1.7.0.4

